src/test/java/com/internship/socialnetwork/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,177 @@
+
+import com.internship.socialnetwork.model.User;
+import com.internship.socialnetwork.repository.UserRepository;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceImplTest {
+
+    private final Long userId1 = 1L;
+
+    private final Long userId2 = 2L;
+
+    private final String username1 = "user1";
+
+    private final String username2 = "user2";
+
+    private final String name1 = "John";
+
+    private final String name2 = "Alice";
+
+    private  final String surname1 = "Doe";
+
+    private final String surname2 = "Johnson";
+
+    private final String email = "test@email.com";
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shouldGetUser() {
+        User user = new User();
+        user.setId(userId1);
+        user.setUsername(username1);
+        when(userRepository.findById(userId1)).thenReturn(Optional.of(user));
+
+        UserDto userDto = userService.getUser(userId1);
+
+        assertNotNull(userDto);
+        assertEquals(username1, userDto.getUsername());
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void shouldThrowNotFoundException_whenGetUser_ifUserDoesNotExist() {
+        when(userRepository.findById(userId2)).thenReturn(Optional.empty());
+
+        userService.getUser(userId2);
+    }
+
+    @Test
+    public void shouldGetAllUsers() {
+        List<User> users = new ArrayList<>();
+
+        users.add(User.builder().username(username1).build());
+        users.add(User.builder().username(username2).build());
+
+        when(userRepository.findAll()).thenReturn(users);
+
+        List<UserDto> userDtos = userService.getAllUsers();
+
+        assertEquals(2, userDtos.size());
+    }
+
+    @Test
+    public void testCreateUser_Success() {
+        UserDto userDto = UserDto.builder()
+                .username(username1)
+                .name(name1)
+                .surname(surname1)
+                .email(email)
+                .build();
+
+        when(userRepository.findByUsernameOrEmail(anyString(), anyString())).thenReturn(Optional.empty());
+
+        User savedUser = User.builder()
+                .username(username1)
+                .name(name1)
+                .surname(surname1)
+                .email(email)
+                .build();
+
+        when(userRepository.save(any(User.class))).thenReturn(savedUser);
+
+        UserDto createdUserDto = userService.createUser(userDto);
+
+        assertEquals(userDto.getUsername(), createdUserDto.getUsername());
+        assertEquals(userDto.getName(), createdUserDto.getName());
+        assertEquals(userDto.getSurname(), createdUserDto.getSurname());
+        assertEquals(userDto.getEmail(), createdUserDto.getEmail());
+    }
+
+    @Test(expected = BadRequestException.class)
```

Comment:
You can also validate exception message, once you use assertThrows with JUnit 5 code